icu4c/source/common/uresbund.cpp

```diff
@@ -489,6 +489,9 @@ findFirstExisting(const char* path, char* name,
 
         /*Fallback data stuff*/
         *hasChopped = chopLocale(name);
+        if (*hasChopped && *name == '\0') {
+            uprv_strcpy(name, "und");
```

Comment:
Looks like this is falling back from "_IN" to "und". I think that's wrong: "und" should be the same as "root", and we want to go through the default locale first.

I think if (we have not yet done the default locale fallback && (hasChopped && name is empty)) we should go directly to the default locale.

Reply:
This is what the code currently at HEAD does:

a)
1. `"und_IN"`
2. `"und"` → load default locale

b)
1. `"_IN"`
2. `""` → load root locale

As I understood the conclusion in the meeting, the latter behaviour is undesired and an input value of `"_IN"` ought to be handled just the same as an input value of `"und_IN"`, ie. after chopping the region code, `"und"` should be requested in order to trigger loading of the default locale.

Therefore, the patch I'm proposing here will change the behaviour to this:

a)
1. `"und_IN"`
2. `"und"` → load default locale

b)
1. `"_IN"`
2. `"und"` → load default locale

Isn't that what we want? Do you think that I have misunderstood something here?

Reply:
We want the effect of "load default locale", but I worry that if there is a und.txt bundle we would open it (and then the root bundle as its parent).

We should recognize that "und" is the same as "" and "root". If we ask to open "und", we should load "root" and nothing else. If we ask to open "und_IN" or "_IN" we should look for "_IN.txt" then default locale then root, and ideally not even try to load a und.txt file.

I realize we are close to the wire, and if it's too much work to skip the parent "und" explicitly, then I can accept your current change -- we would basically rely on there not being a und.txt file.

I also worry that work on ICU-20273 might then break this again.

Reply:
This PR and ICU-20431 is about updating ICU4C to start handling `"_IN"` the same way it already handles `"und_IN"`, without changing anything about how it handles `"und_IN"`.

Your comments about "I worry that if there is a und.txt bundle we would open it" etc. seem to be about how `"und_IN"` is already handled, something which this PR and ICU-20431 does not intend to change.

That's why it was requested to create a separate ticket for this change, to be done before ICU-20273, wasn't it?

Reply:
ok